File: android/src/main/kotlin/dev/volskaya/realtime_audio/Responses.kt
package dev.volskaya.realtime_audio

data class CreateResponse(
  val id: String
) : FlutterResponse() {
  fun toMap(): Map<String, Any> {
    return mapOf("id" to id)
  }
}

class DestroyResponse() : FlutterResponse() {
  fun toMap(): Map<String, Any> {
    return mapOf()
  }
}

data class GetRecorderPermissionResponse(
  val permission: RealtimeAudioRecorderPermission
) : FlutterResponse() {
  fun toMap(): Map<String, Any> {
    return mapOf("permission" to permission.name)
  }
}

data class RequestRecorderPermissionResponse(
  val permission: RealtimeAudioRecorderPermission
) : FlutterResponse() {
  fun toMap(): Map<String, Any> {
    return mapOf("permission" to permission.name)
  }
}

File: darwin/Classes/Responses.swift
struct CreateResponse: FlutterResponse {
  var id: String
}

struct DestroyResponse: FlutterResponse {

}

struct GetRecordPermissionResponse: FlutterResponse {
  var permission: RealtimeAudioRecordPermission
}

struct RequestRecordPermissionResponse: FlutterResponse {
  var permission: RealtimeAudioRecordPermission
}
